tests/unit_tests/test_optional-enhancements.py

```diff
@@ -0,0 +1,347 @@
+
+    # Assert
+    assert len(sam.inventory)== 5
+    assert newest_item is None
+
+
+# @pytest.mark.skip
+def test_get_newest_by_category_with_empty_inventory():
+    # Arrange
+    sam = Vendor(
+        inventory=[]
+    )
+
+    # Act
+    newest_item = sam.get_newest_by_category("Decor")
+
+    # Assert
+    assert len(sam.inventory)== 0
+    assert newest_item is None
+
+
+# @pytest.mark.skip
+def test_get_newest_by_category_no_duplicate():
+    # Arrange
+    item_a = Clothing(age=2)
+    item_b = Decor(age=2)
+    item_c = Clothing(age=4)
+    item_d = Decor(age=2)
+    item_e = Clothing(age=3)
+    sam = Vendor(
+        inventory=[item_a, item_b, item_c, item_d, item_e]
+    )
+
+    # Act
+    newest_item = sam.get_newest_by_category("Decor")
+
+    # Assert
+    assert len(sam.inventory)== 5
+    assert newest_item.category == "Decor"
+    assert newest_item.age == 2
+
+
+# @pytest.mark.skip
+def test_swap_newest_by_category():
+    # Arrange
+    # me
+    item_a = Electronics(age=2)
+    item_b = Decor(age=2)
+    item_c = Electronics(age=4)
+    sunny = Vendor(
+        inventory=[item_a, item_b, item_c]
+    )
+
+    # them
+    item_d = Clothing(age=2)
+    item_e = Decor(age=1)
+    item_f = Decor(age=4)
+    jesse = Vendor(
+        inventory=[item_d, item_e, item_f]
+    )
+
+    # Act
+    result = sunny.swap_newest_by_category(
+        other=jesse,
+        my_priority="Decor",
+        their_priority="Electronics"
+    )
+
+    # Assert
+    # - That the results is truthy
+    assert result is True
+    # - That sunny and jesse's inventories are the correct length
+    assert len(sunny.inventory) == 3
+    assert len(jesse.inventory) == 3
+    # - That all the correct items are in sunny and jesse's inventories, including the items which were swapped from one vendor to the other
+    # test items are in sunny's inventories
+    assert sunny.inventory[0] == item_b
+    assert sunny.inventory[0].category == "Decor"
+    assert sunny.inventory[0].age == 2
+    assert sunny.inventory[1] == item_c
+    assert sunny.inventory[1].category == "Electronics"
+    assert sunny.inventory[1].age == 4
+    assert sunny.inventory[2] == item_e
+    assert sunny.inventory[2].category == "Decor"
+    assert sunny.inventory[2].age == 1
+    # test swapped item is not in the sunny's inventory
+    for item in sunny.inventory:
+        assert item != item_a
+    # test items are in jesse's inventories
+    assert jesse.inventory[0] == item_d
+    assert jesse.inventory[0].category == "Clothing"
+    assert jesse.inventory[0].age == 2
+    assert jesse.inventory[1] == item_f
+    assert jesse.inventory[1].category == "Decor"
+    assert jesse.inventory[1].age == 4
+    assert jesse.inventory[2] == item_a
+    assert jesse.inventory[2].category == "Electronics"
+    assert jesse.inventory[2].age == 2
+    # test swapped item is not in the jesse's inventory
+    for item in jesse.inventory:
+        assert item != item_e
```

Comment:
We could instead write:

```
assert item_e not in jesse.inventory
```

This helps make our test a bit simpler and reduces the amount of logic that happens in the test itself.